src/AutoRest.CSharp/Common/Output/Expressions/KnownValueExpressions/System/ResultExpression.cs

```diff
@@ -3,24 +3,23 @@
 
 using System;
 using System.Net.ClientModel;
-using AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions.Base;
 using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
 using AutoRest.CSharp.Generation.Types;
 
 namespace AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions.System
 {
-    internal sealed record ResultExpression(ValueExpression Untyped) : BaseResponseExpression(typeof(Result), Untyped)
+    internal sealed record ResultExpression(ValueExpression Untyped) : TypedValueExpression<Result>(Untyped)
     {
-        public override ValueExpression Value => Property(nameof(Result<object>.Value));
-        public override BinaryDataExpression Content => throw new InvalidOperationException("Result does not have a Content property");
-        public override StreamExpression ContentStream => throw new InvalidOperationException("Result does not have a ContentStream property");
+        public ValueExpression Value => Property(nameof(Result<object>.Value));
+        public BinaryDataExpression Content => throw new InvalidOperationException("Result does not have a Content property");
+        public StreamExpression ContentStream => throw new InvalidOperationException("Result does not have a ContentStream property");
 
-        public override BaseResponseExpression FromValue(ValueExpression value)
-            => new ResultExpression(new InvokeStaticMethodExpression(typeof(Result), nameof(Result.FromValue), new[] { value, this }));
+        public ResultExpression FromValue(ValueExpression value)
+            => new(new InvokeStaticMethodExpression(typeof(Result), nameof(Result.FromValue), new[] { value, this }));
 
-        public override BaseResponseExpression FromValue(CSharpType explicitValueType, ValueExpression value)
-            => new ResultExpression(new InvokeStaticMethodExpression(typeof(Result), nameof(Result.FromValue), new[] { value, this }, new[] { explicitValueType }));
+        public ResultExpression FromValue(CSharpType explicitValueType, ValueExpression value)
+            => new(new InvokeStaticMethodExpression(typeof(Result), nameof(Result.FromValue), new[] { value, this }, new[] { explicitValueType }));
 
-        public override BaseResponseExpression GetRawResponse() => new ResultExpression(Invoke(nameof(Result<object>.GetRawResponse)));
+        public ResultExpression GetRawResponse() => new(Invoke(nameof(Result<object>.GetRawResponse)));
     }
```

Comment:
I did not see a place in the writer or type provider or method builder that is using this type. Is this expected?

Reply:
It is used in `SystemRestOperationsSnippets`